std/typecons.d

```diff
@@ -4009,6 +4015,24 @@ nothrow pure @nogc @safe unittest
     assert(result.get == 4);
 }
 
+// test that Nullable.get(default) can merge types
+@safe @nogc nothrow pure
+unittest
+{
+    Nullable!ubyte sample = Nullable!ubyte();
+
+    assert(sample.get(1000) == 1000);
```

Comment:
```suggestion
    assert(sample.get(1000) == 1000);
    // Test that `get(U)` return the common type of `T` and `U`
    static assert(is(typeof(sample.get(1000)) == int));
```

Reply:
If it's not `int`, the `get(1000)` call can't return `1000`.

Reply:
I'm sorry I don't understand this comment ?

Reply:
The type is checked implicitly by comparing with a value that's too big for `ubyte`. We don't really care about whether it returns `ubyte` or `int` except in that if we pass in a value that's larger than the data range, then when the Nullable is null, we get the same value back out without truncation.

Reply:
@FeepingCreature is correct, the test is implicitly checking that it's `int`, because `ubyte` can't be 1000. It also captures the semantic expectation -- if it's `null` I want to use 1000, whatever type that means.

Reply:
Fine by me